libgoal/libgoal.go

```diff
@@ -804,8 +808,24 @@ func (c *Client) PendingTransactionInformationV2(txid string) (resp generatedV2.
 	return
 }
 
+// ParsedPendingTransaction takes a txid and returns the parsed PendingTransaction response.
+func (c *Client) ParsedPendingTransaction(txid string) (txn v2.PreEncodedTxInfo, err error) {
```

Comment:
@algochoi Request for clarification:  In what scenarios should a `Client` user prefer `PendingTransactionInformationV2` over `ParsedPendingTransaction`?

* From what I see, it seems `ParsedPendingTransaction` provides equivalent information with a more precise `txn` definition.
* Based on that understanding:  
  * It feels like `ParsedPendingTransaction` is _always_ preferable.
  * I can imagine the client performing a sanity check that calls the same endpoint while varying the codec (msgpack vs json) + confirms responses are equivalent.

Reply:
> From what I see, it seems ParsedPendingTransaction provides equivalent information with a more precise txn definition.

I agree, for internal tests, the precise `txn` definition is better if it is being used.

> I can imagine the client performing a sanity check that calls the same endpoint while varying the codec (msgpack vs json) + confirms responses are equivalent.

To clarify: Do you see this check being done in a separate unit test? Or do the check every time inside the `ParsedPendingTransaction` function?

Reply:
@algochoi Resolving - Thanks for sharing your perspective.  Open to discussing further though I'm comfortable as is for the PR's scope.

>  Do you see this check being done in a separate unit test? Or do the check every time inside the ParsedPendingTransaction function?

* Provided the response does _not_ vary by response encoding (JSON vs msgpack), I think the `RestClient` caller should be oblivious to the response encoding.   A `RestClient` caller expects a 1st class Go representation, _not_ msgpack or JSON.
* For improved coverage, I think it'd be _ideal_ to configure a `RestClient` policy where each exposed method makes N API calls (where N = encoding count).  When making > 1 call, `RestClient` can assert that all responses match and return a single one back to the caller.
* I can imagine adding tests for all encodings elsewhere though the e2e-go tests feels like a convenient place.
* Unfortunately, pending transaction info is _probably_ a poor candidate.  A transaction's state is liable to change across calls.  I hadn't considered the point in my initial note.
  * An alternative is to _randomly_ select an encoding.